src/Fixer/FunctionNotation/PhpdocToReturnTypeFixer.php

```diff
@@ -0,0 +1,325 @@
+        [T_STRING, '__construct'],
+        [T_STRING, '__destruct'],
+        [T_STRING, '__clone'],
+    ];
+
+    /**
+     * @var array
+     */
+    private $versionSpecificTypes = [
+        'void' => 70100,
+        'iterable' => 70100,
+        'object' => 70200,
+    ];
+
+    /**
+     * @var array
+     */
+    private $scalarTypes = [
+        'bool' => true,
+        'float' => true,
+        'int' => true,
+        'string' => true,
+    ];
+
+    /**
+     * @var array
+     */
+    private $skippedTypes = [
+        'mixed' => true,
+        'resource' => true,
+        'null' => true,
+    ];
+
+    /**
+     * @var string
+     */
+    private $classRegex = '/^\\\\?[a-zA-Z_\\x7f-\\xff](?:\\\\?[a-zA-Z0-9_\\x7f-\\xff]+)*(?<array>\[\])*$/';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Takes `@return` annotation of non-mixed types and adjusts accordingly the function signature. Requires PHP >= 7.0.',
+            [
+                new VersionSpecificCodeSample(
+                    '<?php
+
+/** @return \My\Bar */
+function my_foo()
+{}
+',
+                    new VersionSpecification(70000)
+                ),
+                new VersionSpecificCodeSample(
+                    '<?php
+
+/** @return void */
+function my_foo()
+{}
+',
+                    new VersionSpecification(70100)
+                ),
+                new VersionSpecificCodeSample(
+                    '<?php
+
+/** @return object */
+function my_foo()
+{}
+',
+                    new VersionSpecification(70200)
+                ),
+            ],
+            null,
+            'Modifies the signature of functions.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70000 && $tokens->isTokenKindFound(T_FUNCTION);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run after PhpdocScalarFixer.
+        // should be run before ReturnTypeDeclarationFixer, FullyQualifiedStrictTypesFixer.
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('scalar_types', 'Fix also scalar types; may have unexpected behaviour due to PHP bad type coercion system.'))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(false)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 1; 0 < $index; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
```

Comment:
if it is a `lambda` you can continue here as well (as long as PHP doesn't allow return type hinting on `lambda`'s)

Reply:
Lambdas are allowed to have a return type: https://3v4l.org/NmbNN

Reply:
well TIL, thanks :+1: